app/javascript/components/pages/HomePage.jsx

```diff
@@ -1,123 +1,306 @@
+        var start = new Date(location.start_time);
+  //      var startTime = Date.parse(start);
+        var startHours = start.getHours();
+        var startMin = start.getMinutes();
+        var startDay = start.getDate();
+        var startMonth = start.getMonth();
+        var startYear = start.getYear();
+  //      console.log("start time " + startHours + " : " + startMin);
+        var end = new Date(location.end_time);
+        var endHours = end.getHours();
+        var endMin = end.getMinutes();
+        var endDay = end.getDate();
+        var endMonth = end.getMonth();
+        var endYear = end.getYear();
+
+        var startTotalDays = startYear * startMonth + startDay;
+        var endTotalDays = endYear * endMonth + endDay;
+        console.log("start days " + startTotalDays + "end days " + endTotalDays);
+//        var startTotalHours = startTotalDays * startHours;
+//        var endTotalHours = endTotalDays * endHours;
+//        console.log("start hours " + startTotalHours);
+  //      console.log("end hours " + endTotalHours);
+       if(endTotalDays!=startTotalDays){
+          if(startHours > endHours){
+            time_difference = (Math.abs(endTotalDays - startTotalDays) * 24)
+             - Math.abs(endHours - startHours);
+          }
+          else if (startHours == endHours) {
+            time_difference = Math.abs(endTotalDays - startTotalDays) * 24;
+          }
+          else{
+            time_difference = Math.abs(endHours - startHours) +
+             (Math.abs(endTotalDays - startTotalDays) * 24);
+          }
+        }
+        else{
+          if(startHours == endHours){
+            time_difference = 1;
+          }
+          else{
+            time_difference = Math.abs(endHours - startHours);
+          }
+        }
+
+  //      console.log(start);
+  //      console.log(end);
+        console.log("start time " + location.start_time);
+        console.log("end time " + location.end_time);
+        console.log("time difference " + time_difference);
+  //      var start_min = start.getMinutes();
+        count1 += count;
+        tot_time += count * time_difference;
+      }
+/*      var start = new Date(location.start_time);
+      var startTime = Date.parse(start);
+      var startHours = start.getHours();
+      var startMin = start.getMinutes();
+      console.log("start time " + startHours + " : " + startMin);
+      var end = new Date(location.end_time);
+      var endHours = end.getHours();
+      var endMin = end.getMinutes();
+
+      var time_difference = Math.abs(endHours-startHours);
+
+//      console.log(start);
+//      console.log(end);
+
+//      var start_min = start.getMinutes();
+      count1 += count;
+      tot_time += count * time_difference;*/
+   });
+//   return tot_time;
+
+
+}
+//React.useEffect(() => {
+    getTimeTracker();
+//    console.log(tot_time);
+//}, []);
+
+
+
+
+    var CarouselItems = [
+        {
+            name: tot_time + " Hours Meditated Total",

```

Comment:
```suggestion
            name: (Math.round(number * 10) / 10).toFixed(1) + " Hours Meditated Total",

```
round to 1 decimal place